File: server/building/postgres/spf_rightctl/rightctl_initializer/models/policy.py
# -*- coding: utf-8 -*-

from sqlalchemy import Column, String, TIMESTAMP

from db_manager import Base

class Policy(Base):
	__tablename__ = 'policies'
	id = Column(String(100), primary_key=True)
	policy_tid = Column(String(100), nullable=False)

	created_at = Column(TIMESTAMP, nullable=False)
	created_by = Column(String(512), nullable=False)
	updated_at = Column(TIMESTAMP, default=None)
	updated_by = Column(String(512), nullable=False, default="")

File: server/release/node_src/scripts/Authority/db/models/policy.js
/**
 * Authority DB policy model
 * @module  src/scripts/Auhority/db/models/policy
 */

module.exports = function(sequelize) {

    var Sequelize = require('sequelize');

    return sequelize.define('policy', {
        id: {
            type: Sequelize.STRING(100),
            primaryKey: true
        },
        policy_tid: {
            type: Sequelize.STRING(100),
            primaryKey: true
        },
        created_at: {
            type: Sequelize.DATE,
            allowNull: false
        },
        created_by: {
            type: Sequelize.STRING(512),
            allowNull: false
        },
        updated_at: {
            type: Sequelize.DATE,
            allowNull: true
        },
        updated_by: {
            type: Sequelize.STRING(512),
            allowNull: false,
            defaultValue: ''
        }
    }, {
        underscored: true,
        createdAt: "created_at",
        updatedAt: "updated_at"
    });

};
